homeassistant/components/home_connect/binary_sensor.py

```diff
@@ -123,44 +122,49 @@ async def async_setup_entry(
 ) -> None:
     """Set up the Home Connect binary sensor."""
 
-    def get_entities() -> list[BinarySensorEntity]:
+    def get_entities_for_appliance(
+        appliance: HomeConnectApplianceData,
+    ) -> list[BinarySensorEntity]:
         entities: list[BinarySensorEntity] = []
-        for device in entry.runtime_data.devices:
-            entities.extend(
-                HomeConnectBinarySensor(device, description)
-                for description in BINARY_SENSORS
-                if description.key in device.appliance.status
-            )
-            if BSH_DOOR_STATE in device.appliance.status:
-                entities.append(HomeConnectDoorBinarySensor(device))
+        entities.extend(
+            HomeConnectBinarySensor(entry.runtime_data, appliance, description)
+            for description in BINARY_SENSORS
+            if description.key in appliance.status
+        )
+        if StatusKey.BSH_COMMON_DOOR_STATE in appliance.status:
+            entities.append(HomeConnectDoorBinarySensor(entry.runtime_data, appliance))
         return entities
 
-    async_add_entities(await hass.async_add_executor_job(get_entities), True)
+    entities = [
+        entity
+        for appliance in entry.runtime_data.data.values()
+        for entity in get_entities_for_appliance(appliance)
+    ]
+    async_add_entities(entities, True)
 
 
 class HomeConnectBinarySensor(HomeConnectEntity, BinarySensorEntity):
     """Binary sensor for Home Connect."""
 
     entity_description: HomeConnectBinarySensorEntityDescription
 
-    @property
-    def available(self) -> bool:
-        """Return true if the binary sensor is available."""
-        return self._attr_is_on is not None
+    async def _async_event_update_listener(self, event: Event) -> None:
+        """Update status when an event for the entity is received."""
+        self.set_native_value(cast(str | bool, event.value))
+        self.async_write_ha_state()
 
     async def async_update(self) -> None:
```

Comment:
We shouldn't overwrite `async_update` in coordinator entities. It is defined by default and used to force a coordinator refresh via `homeassistant.update_entity` service action.